### DISCUSSION TITLE
Missing Controls file in version 3.7

### DISCUSSION THREAD

**Discussion by Warren-Whitfield** [pelicanmapping/osgearth#2649] [Support Q&A] [answered]
I'm was using version 3.1.0 from vcpkg. I've now updated vcpkg and it seems to be using version 3.7.
In my code I've included osgEarth/Controls, but that file is missing now. What has happend to it?

**Comment by gwaldron** [comment, marked as answer]
You need to enable the CMake option `OSGEARTH_BUILD_LEGACY_CONTROLS_API` to get it.
Look in `vcpkg/ports/osgearth/portfile.cmake` -- you can set it in the `vcpkg_cmake_configure` command.

**Comment by Warren-Whitfield** [comment]
Ok, and then obviously re-install in vcpkg?

**Reply by gwaldron** [reply]
Right.
Or if you are using a vcpkg manifest in your own project, it should (might?) rebuild automatically.

**Comment by Warren-Whitfield** [comment]
OMG, there goes 2 hours :laughing:

**Comment by Warren-Whitfield** [comment]
I have discovered a similar issue with the class MouseCoordsControlFactory. In your ExampleResources file, there is an #if 0 section that the class (and others) are now in. How do I get these to be available again?

**Reply by gwaldron** [reply]
We do not support the old Controls stuff any more, so if you still want to use it it's up to you to integrate it into your app.
In short, you would:
* create a `LabelControl` 
* create a `MouseCoordsTool` with your `LabelControl` and add it to to your View as an event handler.

**Comment by Warren-Whitfield** [comment]
Ok, cool, I'll do that.
So another issue as part of the upgrade to 3.7.1, I get these compiler errors now. Any thoughts please?
![image](https://github.com/user-attachments/assets/08596086-0087-40f6-aee4-5efe4e92a407)

**Reply by gwaldron** [reply]
You can just delete those.
(These used to be plugins and the USE_OSGEARTH_LAYER macro is for incorporating them into a static build.)
